pkg/kubelet/kuberuntime/logs/logs.go

```diff
@@ -456,11 +457,19 @@ func waitLogs(ctx context.Context, id string, w *fsnotify.Watcher, runtimeServic
 			case fsnotify.Write:
 				return true, false, nil
 			case fsnotify.Create:
-				fallthrough
+				return true, true, nil
 			case fsnotify.Rename:
-				fallthrough
+				// waiting new log file to be created after log file renamed during the log rotation
+				// fix issues:
+				// - https://github.com/kubernetes/kubernetes/issues/59902
+				// - https://github.com/kubernetes/kubernetes/issues/28369
+				err := waitFileRecreated(ctx, e.Name)
+				if err != nil {
+					return false, false, err
+				}
+				return true, true, nil
```

Comment:
```suggestion
				return false, false, nil
```

Reply:
The bug still persists even after making these changes.